great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -63,10 +56,45 @@ class SQLDatasourceWarning(UserWarning):
 class ColumnSplitter:
     column_name: str
     method_name: str
-    param_names: Sequence[str]
```

Comment:
So we don't want to serialize this field anymore?
If we do still want this we could use make this a normal field again and put the `param_defaults` logic into an `always=True` validator.
https://docs.pydantic.dev/usage/validators/#validate-always

Reply:
Yes, it's no longer being serialized. I realized that since we've incorporated `method_name` this can be inferred. In a following PR, I when I add different non-date column splitters, the pattern to add new column splitters will be more clear.